src/test/java/software/sigma/internship/controller/StatisticDataControllerTest.java

```diff
@@ -0,0 +1,130 @@
+package software.sigma.internship.controller;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.util.Assert;
+
+import java.time.LocalDate;
+import java.util.Map;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@Sql(scripts = "/scripts/create-statistic-data.sql", executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+@Sql(scripts = "/scripts/clear-statistic-data.sql", executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+
+public class StatisticDataControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Nested
+    public class FindLatest {
+
+        @Test
+        void successful() throws Exception {
+            mockMvc.perform(get("/statistic-data/latest"))
+                    .andDo(print())
+                    .andExpect(status().isOk())
+                    .andExpect(jsonPath("$.id").value(2));
+        }
+    }
+
+    @Nested
+    public class GetDatasetByLossType {
+        @Test
+        void noParam() throws Exception {
+            MvcResult result = mockMvc
+                    .perform(get("/statistic-data/dataset"))
+                    .andExpect(status().isOk())
+                    .andReturn();
+            Map<LocalDate, Integer> map = objectMapper.readValue(result.getResponse().getContentAsString(), new TypeReference<>() {
+            });
+
+            Assert.isTrue(map.size() == 3, "No 3 values in map: " + map);
```

Comment:
You have to use jupiter Assertions for asserts in tests. And for this case assertEquals is more suitable option.